client/src/components/RelatedContainer.jsx

```diff
@@ -9,23 +9,47 @@ export default function RelatedContainer () {
   //related data needed. {category,name,img,star,def price, sale price }
 
   //style data need to search styles find default? to be true and use that data...
-  const [productInfo, setProductInfo] = useState([1,2,3])//DummyData set to 3 indexes
-
+  const [productsInfo, setProductsInfo] = useState([1,2,3])//DummyData set to 3 indexes
+  const dummyId = 1;
   //Method will be need to break up more below
-  const getProductInfo = () => {
-    axios.get('/product/productid')
-      .then((rawData) => {
-        setProductInfo([rawData.data])
-      })
-      .catch((err) => {
-        console.log('Error getting Product...', err)
-      })
+  async function getProductsInfo(Id) {
+
+    const constructedItem = {}
+
+    const relatedArrayData = await axios.get(`/products/${dummyId}/related`)
+    console.log('RelatedArrayData:', relatedArrayData)
+
+    const allProductData = await Promise.all(relatedArrayData.map(itemId => {
```

Comment:
Couple questions I can't seem to figure out just by looking at the code without running it:

1. So are we assuming that you are going to need allProductData as an array to use somewhere later?

2. I'm understanding this as you trying to return an array of promises to promise.all so that they can all resolve together. Does an array get created when you're not returning anything in the axios calls after .map?

3. I think await and Promise.all are redundant. await will make sure that everything to the right of it will finish processing before assigning it to whatever variable you set to the left. Promise.all will make sure that all the promises in the array it is provided will finish, which then provides a single promise that you can then use a .then block to do whatever you want with the resolved promises. I thiiink that adding an await probably doesn't do anything that Promise.all wouldn't do?

4. If i understand correctly, what is going to happen at the end of this async processing, is that the const allProductData is going to hold a single promise with all the resolved values from the Promise.all in an array -- is this the intended functionality? Or perhaps what you want is to create a .then block after the Promise.all and assign the const allProductData the resolved promise value in the .then block

Reply:
I agree with Erik. I'm pretty sure you'd only need to use `promise.all` if you're making several asynchronous requests in parallel (e.g., making the second request before you wait for the first request to resolve), but you need all of them to resolve before your next action. If you're using `.then()` blocks, each one will wait for the previous block to resolve before it executes.